core/src/network.rs

```diff
@@ -578,12 +578,56 @@ pub enum DialError<THandler> {
     },
     /// The dialing attempt is rejected because the peer being dialed is the local peer.
     LocalPeerId { handler: THandler },
+    /// The dialing attempt is rejected because the PeerId is invalid.
     InvalidPeerId {
         handler: THandler,
         multihash: Multihash,
     },
 }
 
+impl<THandler> fmt::Debug for DialError<THandler> {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match self {
+            DialError::ConnectionLimit { limit, handler: _ } => f
+                .debug_struct("DialError::ConnectionLimit")
+                .field("limit", limit)
+                .finish(),
+            DialError::LocalPeerId { handler: _ } => {
+                f.debug_struct("DialError::LocalPeerId").finish()
+            }
+            DialError::InvalidPeerId {
+                handler: _,
+                multihash,
+            } => f
+                .debug_struct("DialError::InvalidPeerId")
+                .field("multihash", multihash)
+                .finish(),
+        }
+    }
+}
+
+impl<THandler> fmt::Display for DialError<THandler> {
```

Comment:
Would it not be easier to use [`thiserror`](https://github.com/dtolnay/thiserror) here?